### ISSUE TITLE
allowCompoundWords does not work well with userWords

### ISSUE THREAD

**Issue by jwang-pr** [streetsidesoftware/vscode-spell-checker#2402]
```
...
    "cSpell.userWords": [
        "id"
    ],
    "cSpell.allowCompoundWords": true,
...
```
**Expected**: no errors, since `parent` is a word and `id` is a user word.
**Actual**:
![image](https://user-images.githubusercontent.com/83189160/206868025-5ef3f0b2-8cf4-48ab-9e80-ec619e548b19.png)

**Comment by Jason3S** [comment]
@jwang-pr,

`allowCompoundWords` only combines words from the same dictionary and won't combine short (2 letter) words because too many mistakes are hidden.

I suggest keeping `allowCompoundWords` `false` and instead adding compoundable words to your dictionary:

```
"cSpell.userWords": ["*id", "parent*", "*code*", "error*", "err+"]
```

This would allow identifiers like `parentid`, `errorcode`, `codeid`, `parentcodeid`.

Symbol meaning:
- `*` - optional compound.
- `+` - only in a compound: `err+` => `errcode` is allowed, but not `err`.
